### PR TITLE
core: compatibility with GHC 8.10.7

### PR DESCRIPTION
**Pull request by dpwiz**

### CODE REVIEW

File: src/Simplex/Chat.hs [hunk 1]
```diff
@@ -2190,11 +2190,11 @@ processChatCommand' vr = \case
       let validName = T.pack $ mkValidName $ T.unpack displayName
       when (displayName /= validName) $ throwChatError CEInvalidDisplayName {displayName, validName}
     assertUserGroupRole :: GroupInfo -> GroupMemberRole -> m ()
-    assertUserGroupRole g@GroupInfo {membership} requiredRole = do
-      when (membership.memberRole < requiredRole) $ throwChatError $ CEGroupUserRole g requiredRole
-      when (memberStatus membership == GSMemInvited) $ throwChatError (CEGroupNotJoined g)
-      when (memberRemoved membership) $ throwChatError CEGroupMemberUserRemoved
-      unless (memberActive membership) $ throwChatError CEGroupMemberNotActive
+    assertUserGroupRole g@GroupInfo {membership = gm@GroupMember {memberRole}} requiredRole = do
```
Comment (epoberezkin) [inline src/Simplex/Chat.hs:2193]: let's not rename membership to gm

File: src/Simplex/Chat/Messages/Batch.hs [hunk 2]
```diff
@@ -10,13 +11,21 @@ module Simplex.Chat.Messages.Batch
...
+#endif
 import qualified Data.ByteString.Lazy as LB
 import Data.Int (Int64)
 import Simplex.Chat.Controller (ChatError (..), ChatErrorType (..))
 import Simplex.Chat.Messages
 
 data MsgBatch = MsgBatch Builder [SndMessage]
+#if MIN_VERSION_bytestring(0,11,1)
   deriving (Show)
+#else
```
Comment (epoberezkin) [inline src/Simplex/Chat/Messages/Batch.hs:25]: this can be just removed, it's not used

File: src/Simplex/Chat/Messages/Batch.hs [hunk 3]
```diff
@@ -10,13 +11,21 @@ module Simplex.Chat.Messages.Batch
 where
 
 import Data.ByteString.Builder (Builder, charUtf8, lazyByteString)
+#if !MIN_VERSION_bytestring(0,11,1)
+import Data.ByteString.Builder (toLazyByteString)
```
Comment (epoberezkin) [inline src/Simplex/Chat/Messages/Batch.hs:15]: and this too can be removed

File: src/Simplex/Chat/Mobile/Shared.hs [hunk 4]
```diff
@@ -16,11 +16,11 @@ type JSONByteString = LB.ByteString
...
 getByteString ptr len = do
   fp <- newForeignPtr_ ptr
-  pure $ BS fp $ fromIntegral len
+  pure $ PS fp 0 (fromIntegral len)
 {-# INLINE getByteString #-}
 
 putByteString :: Ptr Word8 -> ByteString -> IO ()
-putByteString ptr (BS fp len) =
+putByteString ptr (PS fp _ len) =
   withForeignPtr fp $ \p -> memcpy ptr p len
```
Comment (epoberezkin) [inline src/Simplex/Chat/Mobile/Shared.hs:24]:
```suggestion
putByteString ptr (PS fp offset len) =
  withForeignPtr fp $ \p -> memcpy ptr (p `plusPtr` offset) len
```

File: .github/workflows/build.yml [hunk 5]
```diff
@@ -105,7 +112,7 @@ jobs:
           echo "    flags: +openssl"                                        >> cabal.project.local
 
       - name: Install AppImage dependencies
-        if: startsWith(github.ref, 'refs/tags/v') && matrix.os == 'ubuntu-20.04'
+        if: startsWith(github.ref, 'refs/tags/v') && matrix.assert_name && matrix.os == 'ubuntu-20.04'
```
Comment (epoberezkin) [inline .github/workflows/build.yml:115]: typo?

File: .github/workflows/build.yml [hunk 6]
```diff
@@ -121,17 +128,22 @@ jobs:
           echo "    flags: +openssl"      >> cabal.project.local
 
       - name: Unix build CLI
-        id: unix_cli_build
+        id: unix_build
```
Comment (epoberezkin) [inline .github/workflows/build.yml:131]: why rename?

Comment (dpwiz) [reply .github/workflows/build.yml]: for consistency across the build condition here.
nvm, the split wasn't needed.

**Comment by avently** [comment]
I deleted some checks from if statements that are unneeded and specified one more that is useful to speed up the build process a bit

**Pull request merged**